File: ex6classes.py
# ex6: Classes, Python

import numpy as np


class ExampleSix(object):  # Name can diff from file name
    TWENTY_ONE = 21

    def __init__(self, foo, bar):
        self.foo = foo
        self.bar = bar

    def process(self):
        return np.gcd(self.foo, self.bar)

    def check(self, baz):

        if baz[0] == self.TWENTY_ONE:
            print('Twenty-one!')

File: ex6classes.m
% ex6: Classes, MATLAB

classdef ex6classes < handle
    properties (Constant)
        TWENTY_ONE = 21
    end

    properties
        foo
        bar
    end

    methods
        function obj = ex6classes(foo, bar)
            obj.foo = foo;
            obj.bar = bar;
        end

        function baz = run(obj)
            baz = gcd(obj.foo, obj.bar);
        end

        function check(obj, baz)

            if baz(1) == obj.TWENTY_ONE
                disp('Twenty-one!');
            end
        end
    end
end
